File: apps/executive-dashboard/src/types/scoring.ts
/**
 * Scoring Types — exact mirror of schemas/scoring.py
 *
 * Source: services/intelligence-api/app/schemas/scoring.py
 */

import type { LanguageVariant } from "./scenario";

export type ScoreCategory =
  | "macro"
  | "gdp"
  | "country"
  | "sector"
  | "decision"
  | "composite";

export interface ScoreComponent {
  name: string;
  raw_value: number;
  weight: number; // 0-1
  weighted_value: number;
  source_layer: string;
}

export interface ScoreResult {
  id: string;
  category: ScoreCategory;
  label: LanguageVariant;
  score: number; // 0-100
  components: ScoreComponent[];
  confidence: number; // 0-1
  explanation: LanguageVariant;
}

export interface CompositeScore {
  scenario_id: string;
  overall_score: number; // 0-100
  sub_scores: ScoreResult[];
  explanation: LanguageVariant;
  confidence: number; // 0-1
}

File: services/intelligence-api/app/schemas/scoring.py
"""
Scoring schemas — deterministic, weighted, explainable scoring model.

Every score must be: traceable, weighted, reproducible.
"""
from enum import Enum
from pydantic import BaseModel, Field
from .scenario import LanguageVariant


class ScoreCategory(str, Enum):
    MACRO = "macro"
    GDP = "gdp"
    COUNTRY = "country"
    SECTOR = "sector"
    DECISION = "decision"
    COMPOSITE = "composite"


class ScoreComponent(BaseModel):
    """A single scored component with weight and contribution."""
    name: str
    raw_value: float
    weight: float = Field(ge=0.0, le=1.0)
    weighted_value: float
    source_layer: str  # which engine/layer produced this


class ScoreResult(BaseModel):
    """A fully explainable score."""
    id: str
    category: ScoreCategory
    label: LanguageVariant
    score: float = Field(ge=0.0, le=100.0)
    components: list[ScoreComponent]
    confidence: float = Field(ge=0.0, le=1.0)
    explanation: LanguageVariant


class CompositeScore(BaseModel):
    """Executive-level composite score built from multiple ScoreResults."""
    scenario_id: str
    overall_score: float = Field(ge=0.0, le=100.0)
    sub_scores: list[ScoreResult]
    explanation: LanguageVariant
    confidence: float = Field(ge=0.0, le=1.0)
